runtime/win32.c

```diff
@@ -781,6 +781,22 @@ CAMLexport wchar_t *caml_win32_getenv(wchar_t const *lpName)
 
 int caml_win32_rename(const wchar_t * oldpath, const wchar_t * newpath)
 {
+  /* First handle corner-cases not handled by MoveFileEx:
+     - dir to empty dir - positive - should succeed
+     - dir to existing file - should fail */
+  if (GetFileAttributes(oldpath) & FILE_ATTRIBUTE_DIRECTORY) {
```

Comment:
I'm no Windows expert these days, but I did wonder if this check is correct.  If it fails, doesn't it return `-1` and therefore pass?  This [little guide to Windows directory checking](https://www.softwareverify.com/blog/the-correct-way-to-determine-if-a-file-is-a-directory/) seems to say that we should explicitly check for `INVALID_FILE_ATTRIBUTES` (-1) return values, and also for `FILE_ATTRIBUTE_HIDDEN` and `FILE_ATTRIBUTE_SYSTEM`.  I only bring this up because we should be extra careful before executing a file/directory deletion operation...

Reply:
Thanks - that's a nice resource for this Windows newbie! :sweat_smile: 
3856e2b contains a second attempt...